contracts/NodeDelegator.sol

```diff
@@ -338,6 +395,167 @@ contract NodeDelegator is INodeDelegator, LRTConfigRoleChecker, PausableUpgradea
+    /// Is only callable by the `LRTDepositPool` contract.
+    /// @param strategyAddress the address of the EigenLayer LST strategy to withdraw from
+    /// @param strategyShares the amount of EigenLayer strategy shares to redeem
+    /// @param staker the address of the staker requesting the withdrawal.
+    function requestWithdrawal(
+        address strategyAddress,
+        uint256 strategyShares,
+        address staker
+    )
+        external
+        onlyDepositPool
+        whenNotPaused
+    {
+        // request the withdrawal of the LST asset from EigenLayer
+        bytes32 withdrawalRoot = _requestWithdrawal(strategyAddress, strategyShares);
+
+        // store a mapping of the returned withdrawalRoot to the staker withdrawing
+        withdrawalRequests[withdrawalRoot] = staker;
+    }
+
+    /// @notice Requests a withdrawal of liquid staking tokens (LST) from EigenLayer's underlying strategy.
+    /// Must wait `minWithdrawalDelayBlocks` on EigenLayer's `DelegationManager` contract
+    /// before claiming the withdrawal.
+    /// This is currently set to 50,400 blocks (7 days) on mainnet. 10 blocks on Holesky.
+    /// Is only callable by accounts with the Operator role.
+    /// @param strategyAddress the address of the EigenLayer LST strategy to withdraw from
+    /// @param strategyShares the amount of EigenLayer strategy shares to redeem
+    function requestInternalWithdrawal(address strategyAddress, uint256 strategyShares) external onlyLRTOperator {
+        // request the withdrawal of the LSTs from EigenLayer's underlying strategy.
+        _requestWithdrawal(strategyAddress, strategyShares);
+
+        // account for the pending withdrawal as the shares are no longer accounted for in the EigenLayer strategy
+        pendingInternalShareWithdrawals[strategyAddress] += strategyShares;
+    }
+
+    /// @dev request the withdrawal of the LSTs from EigenLayer's underlying strategy.
+    /// @param strategy the address of the EigenLayer LST strategy to withdraw from
+    /// @param strategyShares the amount of EigenLayer strategy shares to redeem
+    /// @return withdrawalRoot the hash of the withdrawal data
+    function _requestWithdrawal(address strategy, uint256 strategyShares) internal returns (bytes32 withdrawalRoot) {
+        IStrategy[] memory strategies = new IStrategy[](1);
+        strategies[0] = IStrategy(strategy);
+
+        // Calculate how many EigenLayer strategy shares to redeem to get the requested asset amount
+        uint256[] memory shares = new uint256[](1);
+        shares[0] = strategyShares;
+
+        // request the withdrawal of the LSTs from EigenLayer
+        IDelegationManager.QueuedWithdrawalParams[] memory requests = new IDelegationManager.QueuedWithdrawalParams[](1);
+        requests[0] = IDelegationManager.QueuedWithdrawalParams(strategies, shares, address(this));
+        address delegationManagerAddress = lrtConfig.getContract(LRTConstants.EIGEN_DELEGATION_MANAGER);
```

Comment:
We should either shorted to `delegationManager`, or create a typed `delegationManager` like we do elsewhere.

Reply:
I make it consistent with the rest of the code and created `delegationManager` of type `IDelegationManager`